File: 4-3/search/binary/binary.py
def binary_search(array, target):
    left, right = 0, len(array)
    
    while left < right:
        mid = left + (right - left) // 2
        
        if array[mid] == target:
            return mid
        elif array[mid] < target:
            left = mid + 1
        else:
            right = mid
            
    return -1 

def main():
    array = [10, 20, 30, 40, 50, 60, 70]
    target = 30

    result = binary_search(array, target)
    print(f"Element found at index {result}" if result != -1 else "Element not found")
    
if __name__ == "__main__":
    main()

File: 4-3/search/binary/go/binary.go
package main

import (
	"fmt"
)

func binarySearch(array []int, target int) int {
	left, right := 0, len(array) - 1

	for left < right {
		mid := left + (right - left) / 2

		if array[mid] == target {
			return mid
		} else if array[mid] < target {
			left = mid + 1
		} else {
			right = mid
		}
	}

	return -1
}

func main() {
	array := []int{10, 20, 30, 40, 50, 60, 70}
	target := 30

	result := binarySearch(array, target)

	if result != -1 {
		fmt.Printf("Element found at index %d\n", result)
	} else {
		fmt.Println("Element not found!")
	}
}